samples/src/main/java/Query.java

```diff
@@ -19,7 +20,15 @@ public static void main(String[] args) {
                     System.getProperty("appId"),
                     System.getProperty("appKey"),
                     System.getProperty("appTenant"));
-            ClientImpl client = new ClientImpl(csb);
+
+            HttpClientProperties properties = HttpClientProperties.builder()
+                    .keepAlive(true)
+                    .maxKeepAliveTime(120)
+                    .maxConnectionsPerRoute(40)
+                    .maxConnectionsTotal(40)
+                    .build();
+
+            ClientImpl client = new ClientImpl(csb, properties);
```

Comment:
This is great to include a sample!

I suggest also adding 1 E2E test, and of course UTs.

Reply:
Looking at the code we added, the factory and the properties class, I don't see what UT we can add for that. Regarding the E2E, there isn't new functionality added in this PR and I think testing should be covered by existing E2Es.

Reply:
I think UTs would be unnecessary for the properties class, but I think it would be important to unit test `create()` in `HttpClientFactory`.

I would recommend an E2E test that injects the `CloseableHttpClient` (you can make that property `protected` in `ClientImpl`, executes 2 commands (e.g. queries), and spy to verify that the same object was used for both calls.

Reply:
@yihezkel added the UT and E2E you mentioned.